src/test/java/org/apache/maven/plugins/dependency/TestGetClassesMojo.java

```diff
@@ -0,0 +1,88 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.LegacySupport;
+import org.apache.maven.plugin.testing.stubs.MavenProjectStub;
+import org.apache.maven.settings.Server;
+import org.apache.maven.settings.Settings;
+import org.sonatype.aether.impl.internal.SimpleLocalRepositoryManager;
+import org.sonatype.aether.util.DefaultRepositorySystemSession;
+
+import java.io.File;
+
+public class TestGetClassesMojo
+        extends AbstractDependencyMojoTestCase
+{
+    GetClassesMojo mojo;
+
+    protected void setUp()
+            throws Exception
+    {
+        // required for mojo lookups to work
+        super.setUp( "markers", false );
+
+        File testPom = new File( getBasedir(), "target/test-classes/unit/get-test/plugin-config.xml" );
+        assert testPom.exists();
+        mojo = (GetClassesMojo) lookupMojo( "get-classes", testPom );
+
+        assertNotNull( mojo );
+
+        LegacySupport legacySupport = lookup( LegacySupport.class );
+        MavenSession session = newMavenSession( new MavenProjectStub() );
+        Settings settings = session.getSettings();
+        Server server = new Server();
+        server.setId( "myserver" );
+        server.setUsername( "foo" );
+        server.setPassword( "bar" );
+        settings.addServer( server );
+        legacySupport.setSession( session );
+        DefaultRepositorySystemSession repoSession =
+                (DefaultRepositorySystemSession) legacySupport.getRepositorySession();
+        repoSession.setLocalRepositoryManager( new SimpleLocalRepositoryManager( testDir.getAbsolutePath() ) );
+
+        setVariableValueToObject( mojo, "session", legacySupport.getSession() );
+    }
+
+    public void testGetClassesNotTransitive()
+            throws Exception
+    {
+        setVariableValueToObject( mojo, "remoteRepositories", "central::default::https://repo.maven.apache.org/maven2,"
+                + "central::::https://repo.maven.apache.org/maven2," + "https://repo.maven.apache.org/maven2" );
+        mojo.setArtifact("org.apache.commons:commons-lang3:3.6");
+        setVariableValueToObject( mojo, "transitive", Boolean.FALSE );
+
+        mojo.execute();
+        return;
+    }
+
+    public void testGetClassesTransitive()
```

Comment:
Um, this does not seem to test anything except that it does not explode.

Reply:
When listing classes for something like org.apache.maven:maven-model:2.0.9, setting -Dtransitive=true lists more classes than the false alternative. It's possible that the artifact used in the test has little/no transitive dependencies which means the results would be similar/the same.